Track Buddy/MotionManager.swift

```diff
@@ -65,13 +68,60 @@ class MotionManager: ObservableObject {
     // Store recent rate-limited points to render path representing past G force values
     private var recentPoints: Deque<CGPoint> = [] // TODO: dluo- think about thread safety?
     
-    func pointPath(atScale factor: CGFloat) -> CGMutablePath {
-        let points = recentPoints.map { CGPoint(x: $0.x * factor, y: $0.y * factor) }
+    func pointPath(at scaleFactor: CGFloat) -> CGMutablePath {
+        
+        /* Interpolate points to fill in gaps between rate-limited points to create a smooth tracer line.
+         
+         Using this interpolate-to-smooth approach because generating the line with the raw accelerometer data at 100 Hz
+         can produce visible divergence between the movement of the dot in the graph and the line that is traced under it.
+         
+         */
+        
+        let interpolatedPoints = interpolate(recentPoints, at: scaleFactor)
         let path = CGMutablePath()
-        path.addLines(between: points)
+        path.addLines(between: interpolatedPoints)
         return path
     }
     
+    private func interpolate(_ points: Deque<CGPoint>, at scaleFactor: CGFloat) -> [CGPoint] {
+        guard points.count > 0 else { return [] }
+        let xPoints = points.map { Double($0.x * scaleFactor) }
+        let yPoints = points.map { Double($0.y * scaleFactor) }
+        let numberOfInterpolatedPoints = Int(Double(1) / Parameters.deviceMotionUpdateInterval * Double(points.count) * graphUpdateInterval)
+        
+        // Generate control array to smooth interpolation result
+        let denominator = Double(numberOfInterpolatedPoints) / Double(points.count - 1)
```

Comment:
Do we have to guard against dividing by 0 if there's only 1 point?

Reply:
I'll change initial `guard` statement to `guard points.count > 1 else { return Array(points) }`